Scripts/capture-build-details.sh

```diff
@@ -5,27 +5,78 @@
 #
 #  Copyright © 2019 LoopKit Authors. All rights reserved.
 
-echo "Gathering build details in ${SRCROOT}"
-cd "${SRCROOT}"
+SCRIPT="$(basename "${0}")"
+SCRIPT_DIRECTORY="$(dirname "${0}")"
 
+error() {
+  echo "ERROR: ${*}" >&2
+  echo "Usage: ${SCRIPT} [-r|--git-source-root git-source-root] [-p|--provisioning-profile-path provisioning-profile-path] [-i|--info-plist-path info-plist-path]" >&2
+  echo "Parameters:" >&2
+  echo "  -r|--git-source-root <git-source-root> root location of git repository to gather information from; optional, defaults to \$WORKSPACE_ROOT if present, otherwise defaults to \$SRCROOT" >&2
+  echo "  -p|--provisioning-profile-path         path to the .mobileprovision provisioning profile file to check for expiration; optional, defaults to \${HOME}/Library/MobileDevice/Provisioning Profiles/\${EXPANDED_PROVISIONING_PROFILE}.mobileprovision" >&2
+  echo "  -i|--info-plist-path                   path to the Info.plist file to modify; optional, defaults to \${BUILT_PRODUCTS_DIR}/\${INFOPLIST_PATH}" >&2
+  exit 1
+}
+
+warn() {
+  echo "WARN: ${*}" >&2
+  exit 0
+}
+
+info() {
+  echo "INFO: ${*}" >&2
+}
+
+git_source_root="${WORKSPACE_ROOT:-${SRCROOT}}"
 plist="${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}"
+profile_path="${HOME}/Library/MobileDevice/Provisioning Profiles/${EXPANDED_PROVISIONING_PROFILE}.mobileprovision"
+xcode_build_version=${XCODE_PRODUCT_BUILD_VERSION:-$(xcodebuild -version | grep version | cut -d ' ' -f 3)}
+while [[ $# -gt 0 ]]
+do
+  arg="$1"
+
+  case $arg in
+    -r|--git-source-root)
+      git_source_root="${2}"
+      shift 2
+      ;;
+    -i|--info-plist-path)
+      plist="${2}"
+      shift 2
+      ;;
+    -p|--provisioning-profile-path)
+      profile_path="${2}"
+      shift 2
+      ;;
+  esac
+done
 
+if [ ${#} -ne 0 ]; then
+  error "Unexpected arguments: ${*}"
+fi
+
+if [ "${git_source_root}" == "" ]; then
+  error "Must provide one of --git-source-root, \$WORKSPACE_ROOT, or \$SRCROOT"
```

Comment:
[nit] Suggest using squiggly brackets here for consistency with the rest of the file.